index.html

```diff
@@ -51,12 +53,89 @@
 			xhr.open('GET', 'db.csv');
 			xhr.send();
 		}
+		function alphabet_search_bar() {
+			var alphabetSearchDiv = document.getElementById("alphabet-search");
+
+			// Create a table for the alphabet-search
+			var table = document.createElement("table");
+			var tbody = document.createElement("tbody");
+			var row = document.createElement("tr");
+
+			// Create "ALL" before the alphabet for better access
+			var cell_all = document.createElement("td")
+			var link_all = document.createElement("a");
+			link_all.href = "#"
+			var letter = 'All'
+			link_all.onclick = function (letter) {
+				return function () {
+					searchByAlphabet(letter);
+					console.log(letter)
+				};
+			} (letter)
+			link_all.textContent = letter
+			cell_all.appendChild(link_all)
+			row.appendChild(cell_all)
+
+			// Create a cell for each character
+			for (var i = 65; i <= 90; i++) { // ASCII codes for A-Z
+				var letter = String.fromCharCode(i);
+
+				// Create a table cell
+				var cell = document.createElement("td");
+
+				// Create a link with the specified href and onclick attributes
+				var link = document.createElement("a");
+				link.href = "#";
+				link.onclick = function (letter) {
+					return function () {
+						searchByAlphabet(letter);
+					};
+				}(letter); // Use a closure to capture the current value of 'letter'
+
+				// Set the text content of the link to the current letter
+				link.textContent = letter;
+
+				// Append the link to the cell
+				cell.appendChild(link);
+
+				// Append the cell to the row
+				row.appendChild(cell);
+			}
+
+			// Append the row to the tbody
+			tbody.appendChild(row);
+
+			// Append the tbody to the table
+			table.appendChild(tbody);
+
+			// Append the table to the alphabetSearchDiv
+			alphabetSearchDiv.appendChild(table);
+
+
+		}
+		// function that passes the letter to searching Regex
+		function searchByAlphabet(letter) {
```

Comment:
why do you need a js function to change the hash uppon click ? can't you just put the letter in the while creating the link ?

Reply:
Could you please explain this? I tried to implement this, but couldn't understand. It would be a great help if you could guide me a bit on how could I change the approach the way you are asking me...

Reply:
sure, in HTML you can simply do
```html
<a href="#hello">test</a>
```
this will display a link that, once clicked, will change the hash to #hello